vibe_code_apps_22/ml/interval15.py

```
@@ -0,0 +1,215 @@
+    q = quarter_from_interval_end_hms(end_local.hour, end_local.minute)
+    sd = site_date_for_interval_end(end_local)
+    return Interval15(
+        interval_start_utc=start_utc,
+        interval_end_utc=end_utc,
+        site_date=sd,
+        quarter_index=q,
+    )
+
+
+def hour_ending_int_1_24(q: int) -> int:
+    """Hour bucket 1..24 containing quarter ``q`` (q0–q3 → 1, …, q92–q95 → 24)."""
```

Ruff (0.16.1)

[warning] Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)